kodein-android/src/main/java/com/github/salomonbrys/kodein/android/KodeinAndroidComponents.kt

```diff
@@ -266,6 +266,15 @@ abstract class KodeinAppCompatActivity : AppCompatActivity(), AppCompatActivityI
     }
 }
 
+private fun validateFragmentsActivity(activity: Activity) {
+    if(activity !is KodeinActivity && activity !is KodeinFragmentActivity && activity !is KodeinAppCompatActivity
```

Comment:
Maybe it is simpler to only check if `activity !is AndroidInjector` because all class or interfaces you check inherit from AndroidInjector